sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -245,6 +259,30 @@ private final class SemaphoreTaskInfo() extends Logging {
     }
   }
 
+  def tryAcquire(semaphore: Semaphore): Boolean = synchronized {
+    val t = Thread.currentThread()
+    if (hasSemaphore) {
+      blockedThreads.add(t)
+      moveToActive(t)
```

Comment:
It seems like this could be simplified:
```suggestion
      activeThreads.add(t)
```